File: Codigos estudiantes por lenguaje/Java/Daniel Cangrejo/Ordenamientos/cocktail_sort.java
//cocktail_sort
public class Ordenamientos2_2 {
	
	static void cocktailSort(int a[]) {
        boolean swapped = true;
        int start = 0;
        int end = a.length;
        while (swapped == true) {
            swapped = false;
            for (int i = start; i < end - 1; i++) {
            	//cambiar ">" aqui  si lo quiero de mayor a menor
            	if (a[i] > a[i + 1]) {
                    int temp = a[i];
                    a[i] = a[i + 1];
                    a[i + 1] = temp;
                    swapped = true;
                }
            }
            if (swapped == false) {
                break;
            }
            swapped = false;
            end -= 1;
            for (int i = end - 1; i >= start; i--) {
               //cambiar ">" aqui tambien si lo quiero de mayor a menor
            	if (a[i] > a[i + 1]) {
                    int temp = a[i];
                    a[i] = a[i + 1];
                    a[i + 1] = temp;
                    swapped = true;
                }
            }
            start += 1;
        }
    }

    static void printArray(int a[]) {
        int n = a.length;
        for (int i = 0; i < n; i++) {
            System.out.print(a[i] + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        int a[] = {5, 1, 4, 2, 8, 0, 2};
        printArray(a);
        cocktailSort(a);
        printArray(a);
    }
}

File: Codigos estudiantes por lenguaje/PY/Daniel Cangrejo/Ordenamientos/cocktail_sort.py
from sys import stdin, stdout
rl = stdin.readline
wr = stdout.write


def cocktailSort(arr):

    swapped = True
    start = 0
    end = len(arr)

    while swapped:  # Sort Izq a Der
        swapped = False
        for i in range(start, end-1):
            if arr[i] > arr[i+1]:  # Cambiar < o >
                arr[i], arr[i+1] = arr[i+1], arr[i]
                swapped = True
        if not swapped:
            break

    swapped = False
    end -= 1
    for i in range(end-1, start-1, -1):  # Sort Der a Izq
        if arr[i] > arr[i+1]:  # Cambiar < o >
            arr[i], arr[i+1] = arr[i+1], arr[i]
            swapped = True

    start += 1


def printArray(arr):
    for i in arr:
        wr(f'{i} ')
    wr('\n')


arr = list(map(int, rl().split()))
printArray(arr)
cocktailSort(arr)
printArray(arr)
